app/src/userpages/components/ProfilePage/DeleteAccount/index.jsx

```diff
@@ -0,0 +1,95 @@
+// @flow
+
+import React from 'react'
+import { connect } from 'react-redux'
+import { Translate, I18n } from 'react-redux-i18n'
+import { Button } from 'reactstrap'
+
+import DeleteAccountDialog from './DeleteAccountDialog'
+import { deleteUserAccount } from '$shared/modules/user/actions'
+import { selectDeletingUserAccount } from '$shared/modules/user/selectors'
+import type { StoreState } from '$shared/flowtype/store-state'
+
+import styles from './deleteAccount.pcss'
+
+type State = {
+    modalOpen: boolean,
+}
+
+type StateProps = {
+    deletingUserAccount: boolean,
+}
+type DispatchProps = {
+    deleteAccount: () => Promise<null>,
+}
+
+type Props = StateProps & DispatchProps
+
+class DeleteAccount extends React.Component<Props, State> {
+    state = {
+        modalOpen: false,
+    }
+
+    openModal = () => {
+        this.setState({
+            modalOpen: true,
+        })
+    }
+
+    onClose = () => {
+        this.setState({
+            modalOpen: false,
+        })
+    }
+
+    onSave = () => {
+        this.props.deleteAccount()
+            .then(() => {
+                this.onClose()
+            }, (error) => {
+                alert(error.message) // eslint-disable-line no-alert
```

Comment:
Similar to my previous comment, would also be good to *ensure* `alert` calls never make it into production. Perhaps `no-alert` could be downgraded to a warning so this `eslint-disable` isn't required. Could selectively upgrade warnings to errors specifically for the production build, but I realise this is somewhat complicated by having staging "production", where we'd still want them as warnings, and production "production" where we'd want them as errors.

Reply:
I think the alert itself is not the problem but the missing functionality. In this case it's waiting for the API endpoint implementation although this is missed if there is no TODO comment here.